x-pack/legacy/plugins/monitoring/server/lib/elasticsearch/verify_monitoring_auth.js

```diff
@@ -38,19 +38,29 @@ export async function verifyMonitoringAuth(req) {
 async function verifyHasPrivileges(req) {
   const { callWithRequest } = req.server.plugins.elasticsearch.getCluster('monitoring');
 
-  const response = await callWithRequest(req, 'transport.request', {
-    method: 'POST',
-    path: '/_security/user/_has_privileges',
-    body: {
-      index: [
-        {
-          names: [INDEX_PATTERN], // uses wildcard
-          privileges: ['read'],
-        },
-      ],
-    },
-    ignoreUnavailable: true, // we allow 404 incase the user shutdown security in-between the check and now
-  });
+  let response;
+  try {
+    response = await callWithRequest(req, 'transport.request', {
+      method: 'POST',
+      path: '/_security/user/_has_privileges',
+      body: {
+        index: [
+          {
+            names: [INDEX_PATTERN], // uses wildcard
+            privileges: ['read'],
+          },
+        ],
+      },
+      ignoreUnavailable: true, // we allow 404 incase the user shutdown security in-between the check and now
+    });
+  } catch (err) {
+    if (
+      err.message === 'no handler found for uri [/_security/user/_has_privileges] and method [POST]'
```

Comment:
Nitpicking: Not really a fan of explicit `message` conditions (since they might change). Is there any way to use status code instead? Something like `err.status` or `err.statusCode` (not sure which one it is)

Reply:
I'd love to, but I don't think we can. 

See https://github.com/elastic/kibana/pull/53306/files#r380773669

Reply:
Ah, makes sense now. Thanks for the link 👍